src/main/java/au/com/dmg/fusion/response/diagnosisresponse/DiagnosisResponse.java

```diff
@@ -0,0 +1,118 @@
+package au.com.dmg.fusion.response.diagnosisresponse;
+
+import au.com.dmg.fusion.request.paymentrequest.extenstiondata.ExtensionData;
+import au.com.dmg.fusion.response.HostStatus;
+import au.com.dmg.fusion.response.POIStatus;
+import au.com.dmg.fusion.response.Response;
+import au.com.dmg.fusion.response.ResponseType;
+import com.squareup.moshi.Json;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.List;
+
+public class DiagnosisResponse implements ResponseType {
+
+    @Json(name = "Response")
+    private final Response response;
+    @Json(name = "LoggedSaleID")
+    public List<String> loggedSaleID;
+    @Json(name = "POIStatus")
+    public POIStatus poiStatus;
+    @Json(name = "HostStatus")
+    public List<HostStatus> hostStatus;
+    @Json(name = "ExtensionData")
+    public ExtensionData extensionData;
```

Comment:
This should refer to another _ExtensionData_ class named "**ResponseExtensionData**", which contains **POIInformation**.